### PR TITLE
[fix](repeat)remove unmaterialized expr from repeat node

### PR DESCRIPTION
**Pull request by starocean999**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 37.82 seconds
 load time: 613 seconds
 storage size: 17184113351 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221103155340_clickbench_pr_39510.html

File: fe/fe-core/src/main/java/org/apache/doris/analysis/GroupingInfo.java [hunk 1]
```diff
@@ -89,7 +89,30 @@ public List<Expr> getPreRepeatExprs() {
...
+            }
+        }
+        preRepeatExprs = materializedPreRepeatExprs;
+
+        // remove unmaterialized slotRef from outputTupleSmap and outputTupleDesc if there is any
+        for (Expr expr : unMaterializedSlotRefs) {
+            Expr rExpr = outputTupleSmap.get(expr);
+            outputTupleSmap.removeByRhsExpr(rExpr);
+            if (rExpr instanceof SlotRef) {
+                outputTupleDesc.getSlots().remove(((SlotRef) rExpr).getDesc());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/GroupingInfo.java:113]: is it better to set it as not materialized instead of remove it?

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/analysis/GroupingInfo.java (hunk 1 above)
Comment (starocean999) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/GroupingInfo.java]: set unmaterialized is better, code is updated, thx

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**